tools/codegen/api/translate.py

```diff
@@ -222,6 +226,10 @@ def direct_solve(goal: NamedCType) -> str:
                 intArrayRef_ctype = NamedCType(goal.name, BaseCType(intArrayRefT))
                 argname = direct_solve(intArrayRef_ctype)
                 return f'{argname}.vec()'
+            elif goal.type == OptionalCType(VectorCType(BaseCType(longT))):
+                intArrayRef_ctype = NamedCType(goal.name, BaseCType(optionalIntArrayRefT))
+                argname = direct_solve(optionalIntArrayRef_ctype)
+                return f'{argname}.has_value() ? c10::OptionalIntArrayRef({argname}) : c10::OptionalIntArrayRef'
```

Comment:
@kurtamohler According to the comment above this should return a value type, so I think this should be 
```suggestion
                return f'{argname}.has_value() ? c10::make_optional({argname}.vec()) : c10::nullopt'
```